gtwrap/xml_parser/xml_parser.py

```diff
@@ -0,0 +1,198 @@
+import xml.etree.ElementTree as ET
+import sys
+import os
+
+"""Parses and extracts docs from Doxygen-generated XML.
+"""
+class XMLDocParser:
+    def __init__(self):
+        # Memory for overloaded functions with identical parameter name sets
+        self._memory = {}
+        # This is useful for investigating functions that cause problems for extract_docstring.
+        # Set this to true to have useful information for debugging this class, as in the CLI
+        # function at the bottom of this class.
+        self._verbose = False
+
+    """Get the ElementTree of an XML file given the file name.
+    If an error occurs, prints a warning and returns None.
+    """
+    def parse_xml(self, xml_file: str):
+        try:
+            return ET.parse(xml_file)
+        except FileNotFoundError:
+            print(f"Warning: XML file '{xml_file}' not found.")
+            return None
+        except ET.ParseError:
+            print(f"Warning: Failed to parse XML file '{xml_file}'.")
+            return None
+
+    """Extract the docstrings for a C++ class's method from the Doxygen-generated XML.
+    
+    Parameters:
+        xml_folder: the path to the folder that contains all of the Doxygen-generated XML.
+        cpp_class: the name of the C++ class that contains the function whose docstring is to be extracted.
+        cpp_method: the name of the C++ method whose docstring is to be extracted.
+        method_args_names: a list of the names of the cpp_method's parameters.
+    """
+    def extract_docstring(self, xml_folder: str, cpp_class: str, cpp_method: str, method_args_names: 'list[str]'):
+        if self._verbose: print(f"Extracting docs for {cpp_class}.{cpp_method}")
+
+        # Create the path to the Doxygen XML index file.
+        xml_index_file = xml_folder + os.sep + "index.xml"
+
+        # Parse the index file
+        index_tree = self.parse_xml(xml_index_file)
+        if not index_tree: 
+            if self._verbose: print(f"Index file {xml_index_file} was empty.")
+            return ""
+
+        index_root = index_tree.getroot()
+
+        # Find the compound with name == cpp_class
+        class_index = index_root.find(f"./*[name='{cpp_class}']")
+
+        if class_index is None:
+            if self._verbose: print(f"Could not extract docs for {cpp_class}.{cpp_method}; class not found in index file.")
```

Comment:
Same as above. A better way to do this would be to add a method `maybe_print` which is something like
```python
def maybe_print(self, s: str):
	if self._verbose:
		print(s)
```
and then just call `self.maybe_print(f"some text")`.